src/lazy/raw_value_ref.rs

```diff
@@ -24,6 +25,28 @@ pub enum RawValueRef<'data, D: LazyDecoder<'data>> {
     Struct(D::Struct),
 }
 
+// Provides equality for scalar types, but not containers.
+impl<'data, D: LazyDecoder<'data>> PartialEq for RawValueRef<'data, D> {
+    fn eq(&self, other: &Self) -> bool {
+        use RawValueRef::*;
+        match (self, other) {
+            (Null(i1), Null(i2)) => i1 == i2,
+            (Bool(b1), Bool(b2)) => b1 == b2,
+            (Int(i1), Int(i2)) => i1 == i2,
+            (Float(f1), Float(f2)) => f1 == f2,
+            (Decimal(d1), Decimal(d2)) => d1 == d2,
+            (Timestamp(t1), Timestamp(t2)) => t1 == t2,
+            (String(s1), String(s2)) => s1 == s2,
+            (Symbol(s1), Symbol(s2)) => s1 == s2,
+            (Blob(b1), Blob(b2)) => b1 == b2,
+            (Clob(c1), Clob(c2)) => c1 == c2,
+            // We cannot compare lazy containers as we cannot guarantee that their complete contents
+            // are available in the buffer. Is `{foo: bar}` equal to `{foo: b`?
```

Comment:
> What is the motivation for implementing `PartialEq`?

It allows us to test for equality using `assert_eq!` in unit tests.

> If there was some way of having 3-value logic here (true, false, unknown) that would make me a little more comfortable. I don't like how this would say that {foo: bar} and {foo: bar} are not equal.

This implementation is leaning on the `Partial` part of [`PartialEq`](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html). Just as there are values in `f64` that cannot be compared and always return `false` (i.e. `NaN`), there are values in `RawValueRef` that cannot be compared and always return `false` (i.e. container types).

Let me know if that makes it feel less yucky.

Reply:
Sounds good.